frontend/src/metabase/nav/components/AppBar/AppBarToggle.styled.tsx

```diff
@@ -1,13 +1,36 @@
 import styled from "@emotion/styled";
+import { css } from "@emotion/react";
 import { color } from "metabase/lib/colors";
 import Icon from "metabase/components/Icon";
 
-export const SidebarButton = styled.button`
+interface SidebarButtonProps {
+  isLogoVisible: boolean;
+}
+
+export const SidebarButton = styled.button<SidebarButtonProps>`
   cursor: pointer;
   display: block;
+
+  && {
```

Comment:
need to match the CSS specificity from the original declaration. Otherwise, this rule will always win (nested component selector will result in `.component.nested-component`):
https://github.com/metabase/metabase/blob/9125f5d2e92c81464eb38f881ae1a04a5c86c944/frontend/src/metabase/nav/components/AppBar/AppBarLarge.styled.tsx#L33-L35

Reply:
Could you refactor (move to this component) this too https://github.com/metabase/metabase/blob/9125f5d2e92c81464eb38f881ae1a04a5c86c944/frontend/src/metabase/nav/components/AppBar/AppBarLarge.styled.tsx#L34?

Reply:
At first, I wanted to move both
https://github.com/metabase/metabase/blob/9125f5d2e92c81464eb38f881ae1a04a5c86c944/frontend/src/metabase/nav/components/AppBar/AppBarLarge.styled.tsx#L33-L35
and
https://github.com/metabase/metabase/blob/9125f5d2e92c81464eb38f881ae1a04a5c86c944/frontend/src/metabase/nav/components/AppBar/AppBarLarge.styled.tsx#L43-L45

to here (frontend/src/metabase/nav/components/AppBar/AppBarToggle.styled.tsx).

I think both declarations should stay close to each other since they're related.

But the latter declaration is under `:hover { }`, which means I need to import `AppBarLeftContainer` and nested it in this file instead, so I thought if I couldn't remove the nested component selector maybe I don't do anything at all. That's why I didn't move any of them.

However, as you have pointed out, if we move the first declarations, then we could remove the `&&` hack. But I would still want to move the other hover declaration there too. Are fine with having nested  `AppBarLeftContainer` inside this component? I don't think it's worse as we just move the place where nested component selector lives, but we can remove the `&&` hack.

What do you think?